src/lib/operators/projection.cpp

```diff
@@ -272,11 +272,12 @@ std::shared_ptr<const Table> Projection::_on_execute() {
       std::vector<SortColumnDefinition> transformed;
       transformed.reserve(sorted_by.size());
       for (const auto& [column_id, mode] : sorted_by) {
-        if (!input_column_to_output_column.count(column_id)) {
-          continue;  // column is not present in output expression list
+        // Full iteration as input columns can be projected repeatedly
+        for (const auto& [output_column_id, input_column_id] : output_column_to_input_column) {
+          if (column_id == input_column_id) {
```

Comment:
Any reason you are not using find_if here?

Reply:
I do not only care about the first hit. Another output column might be sorted as well with the sorting information of the same input column.

Reply:
Wouldn't it make sense to swap the two for loops then and to use `find_if`? I won't block the PR either way.

Reply:
Ah, got you. Changed. Thanks.